core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -1018,7 +1018,7 @@ public void performOn(ShardRouting replica, ReplicaRequest request, ActionListen
             }
             transportService.sendRequest(node, transportReplicaAction,
                 new ConcreteShardRequest<>(request, replica.allocationId().getId()), transportOptions,
-                new ActionListenerResponseHandler<>(listener, ReplicaResponse::new));
+                new ActionListenerResponseHandler<ReplicaResponse>(listener, ReplicaResponse::new));
```

Comment:
I think that at a minimum this needs a comment explaining why, otherwise someone is going to see it and think "I'll just diamond operator that up".